Readme

   VIA  -- Volumetric Image Analysis
   

VIA is an image processing package.
It contains about 70 library functions and 50
application programs.

The data format, file-handling routines,
as well as some image processing routines
are taken from the public-domain-package "Vista"
(www.cs.ubc.ca/labs/lci/vista/vista/html).

To compile VIA do the following:
(These are instructions to do an out-of-source build
with the cmake tool. For further information consult the
cmake introductionary pages on 
www.cmake.org/cmake/help/runningcmake.html)

1. create a directory 'build' with 'mkdir build'

2. chdir to the new directory with 'cd build'

3. type cmake ../

4. type 'make'

5. type 'make install'

6. optionally, type 'make clean'




G.Lohmann, MPI-CBS, April 2009.


debian/changelog
via (2.0.4-4) unstable; urgency=low

  * changing build deps vor libcblas on debian testing

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Thu, 20 Jul 2012 09:39:21 +0400
 
via (2.0.4-3) unstable; urgency=low

  * changed build deps

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Fri, 4 May 2012 14:27:00 +0400
 
via (2.0.4-2) unstable; urgency=low

  * Add missing dependency x11proto-core-dev to libvia-dev.

 -- Michael Hanke <mih@debian.org>  Thu, 16 Feb 2012 21:54:50 +0100

via (2.0.4-1) unstable; urgency=low

  * New release.

 -- Michael Hanke <mih@debian.org>  Wed, 08 Feb 2012 17:38:27 +0100

via (2.0.3+svn272-0) UNRELEASED; urgency=low

  * fixed bug in VFIllImage

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 29 Nov 2011 12:17:00 +0400

via (2.0.2+svn207-0) UNRELEASED; urgency=low

  * fixed range of VLong

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 05 May 2011 12:17:00 +0400

via (2.0.2+svn206-1) UNRELEASED; urgency=low

  * New upstream release. Changes in ABI, upstream skipped SO version 1 in
    public releases and moved directly to 2. Merged missing_header patch with
    upstream.
  * Fix via-bin archive section (libs -> science).
  * Package now needs libcblas.so, hence build-depends on libatlas.

 -- Michael Hanke <mih@debian.org>  Thu, 05 May 2011 09:53:53 -0400

via (2.0.1+svn191-0) UNRELEASED; urgency=low

  * renamed via-bin to via-utils

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400

via (2.0.0-0) UNRELEASED; urgency=low

  * This version comes with the new lipsia-2.0 package
    and comprises some minor bug fixes
  * Fixed type error of VLong (is now long)

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400

via (1.6.0-3) unstable; urgency=low

  * Adjust VCS information for the new repository home debian-med.
  * Switch to source package format 3.0 (quilt). Move patches into separate
    series.
  * Adjust build-dep from atlas to libblas-dev (Closes: #554814). Thanks to
    Riku Voipio for reporting.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 06 Apr 2010 11:10:11 -0400

via (1.6.0-2) unstable; urgency=low

  * Apply patch to fix an implicit pointer conversion (Closes: #540292).
    Thanks to Dann Frazier.
  * Update Standards-version to 3.8.3.0 -- no changes necessary.

 -- Michael Hanke <michael.hanke@gmail.com>  Mon, 24 Aug 2009 09:39:41 -0400

via (1.6.0-1) unstable; urgency=low

  * Initial release (Closes: #521328).
  * New Upstream Version. Renamed 'vcat' to 'vcatobj' to not conflict with
    atfs package in Debian.
  * Update Standards-version to 3.8.2.0 -- no changes necessary.
  * Update watchfile for recent change in upstream naming scheme.
  * Add versioned dependency on debhelper (>= 5) upon recommendation by
    lintian.

 -- Michael Hanke <michael.hanke@gmail.com>  Fri, 19 Jun 2009 13:50:52 +0200

via (1.5.2+svn3529-2) UNRELEASED; urgency=low

  * Add alternative build-dependency on atlas3-base-dev to be able to build
    the package on Debian etch.
  * FTP access to the upstream sources needs passive mode. Fixed debian/watch.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 14 Apr 2009 02:46:10 +0200

via (1.5.2+svn3529-1) UNRELEASED; urgency=low

  * New Upstream SVN snapshot that relocates the VIA lib headers to their new
    cannonical location.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 31 Mar 2009 22:08:01 +0200

via (1.5.2-1) UNRELEASED; urgency=low

  * Initial Debian packaging. This is a complete rewrite of the original
    Debian packaging effort done by the upstream developers.

 -- Michael Hanke <michael.hanke@gmail.com>  Thu, 26 Mar 2009 19:46:09 +0100


viaio/Type.c
/*
 *  $Id: Type.c 3177 2008-04-01 14:47:24Z karstenm $
 *
 *  Routines in this file allow one to register types (e.g., "image") along
 *  with sets of routines for operating on objects of those types.
 */

/*
 *  Copyright 1993, 1994 University of British Columbia
 *
 *  Permission to use, copy, modify, distribute, and sell this software and its
 *  documentation for any purpose is hereby granted without fee, provided that
 *  the above copyright notice appears in all copies and that both that
 *  copyright notice and this permission notice appear in supporting
 *  documentation. UBC makes no representations about the suitability of this
 *  software for any purpose. It is provided "as is" without express or
 *  implied warranty.
 *
 *  Author: Arthur Pope, UBC Laboratory for Computational Intelligence
 */

/* From the Vista library: */
#include "viaio/Vlib.h"
#include "viaio/os.h"
#include "viaio/VList.h"
#include "viaio/mu.h"

/* File identification string: */
VRcsId ("$Id: Type.c 3177 2008-04-01 14:47:24Z karstenm $");


/*
 *  Information about built-in types.
 */

extern VTypeMethods VEdgesMethods;	/* in EdgesType.c */
extern VTypeMethods VImageMethods;	/* in ImageType.c */
extern VTypeMethods VGraphMethods;      /* in GraphType.c */
extern VTypeMethods VolumesMethods;     /* in VolumesType.c */

static VRepnInfoRec builtin_repn_info[] = {
    { "unknown" },

    /* Integer and floating-point numbers: */
    { "bit",	   sizeof (VBit),	1, 0.0, 1.0 },
    { "ubyte",	   sizeof (VUByte),	8, 0.0, 255.0 },
    { "sbyte",	   sizeof (VSByte),	8, -128.0, 127.0 },
    { "short",	   sizeof (VShort),    16, -32768.0, 32767.0 },
    { "long",	   sizeof (VLong),     sizeof( VLong ) * 8, -2147483648.0, 2147483647.0 },
    { "float",	   sizeof (VFloat),    32, 
	  -3.40282346638528860e+38, 3.40282346638528860e+38 },
    { "double",	   sizeof (VDouble),   64,
	  -1.797693134862315708e+308, 1.797693134862315708e+308 },

    /* Miscellaneous representations: */
    { "attr-list", sizeof (VAttrList),  0, 0.0, 0.0 },
    { "boolean",   sizeof (VBoolean),	1, 0.0, 0.0 },
    { "bundle",	   sizeof (VPointer),	0, 0.0, 0.0 },
    { "list",	   sizeof (VList),	0, 0.0, 0.0 },
    { "pointer",   sizeof (VPointer),	0, 0.0, 0.0 },
    { "string",	   sizeof (VString),	0, 0.0, 0.0 },

    /* Standard object types: */
    { "edges",	   sizeof (VPointer),	0, 0.0, 0.0, & VEdgesMethods },
    { "image",	   sizeof (VPointer),	0, 0.0, 0.0, & VImageMethods },
    
    /* new object types */
    { "graph",     sizeof (VPointer),   0, 0.0, 0.0, & VGraphMethods },
    { "volumes",   sizeof (VPointer),   0, 0.0, 0.0, & VolumesMethods }
};

VRepnInfoRec *VRepnInfo = builtin_repn_info;
static VRepnKind nRepnKinds = VNRepnKinds;


/*
 *  VRegisterType
 *
 *  Register some handlers for dealing with objects of a particular type.
 *  Returns the VRepnKind code assigned the new type.
 */

VRepnKind VRegisterType (VStringConst name, VTypeMethods *methods)
{
    VRepnInfoRec *p;

    /* Move the existing type information into a bigger table: */
    if (VRepnInfo == builtin_repn_info) {
	VRepnInfo = VMalloc ((VNRepnKinds + 1) * sizeof (VRepnInfoRec));
	VCopy (builtin_repn_info, VRepnInfo, VNRepnKinds);
    } else
	VRepnInfo =
	    VRealloc (VRepnInfo, (nRepnKinds + 1) * sizeof (VRepnInfoRec));

    /* Write the new type's info into the last table entry: */
    p = VRepnInfo + nRepnKinds;
    p->name = VNewString (name);
    p->size = p->precision = p->min_value = p->max_value = 0.0;
    p->methods = methods;

    return nRepnKinds++;
}


/*
 *  VLookupType
 *
 *  Locate information about a named type.
 */

VRepnKind VLookupType (VStringConst name)
{
    VRepnKind repn;

    for (repn = VUnknownRepn; repn < nRepnKinds; repn++)
	if (strcmp (VRepnInfo[repn].name, name) == 0)
	    return repn;
    return VUnknownRepn;
}
